src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2514,6 +2534,21 @@ Node TheoryStringsRewriter::rewriteReplace( Node node ) {
       Node nn2 = node[2].substitute(
           emptyNodes.begin(), emptyNodes.end(), substs.begin(), substs.end());
 
+      // If the contains rewrites to a conjunction of empty-string equalities
+      // and we are doing the replacement in an empty string, we can rewrite
+      // the string-to-replace with a concatenation of all the terms that must
+      // be empty:
+      //
+      // (str.replace "" y z) ---> (str.replace "" (str.++ y1 ... yn)  z)
+      // if (str.contains "" y) ---> (and (= y1 "") ... (= yn ""))
+      if (node[0] == empty && allEqs)
+      {
+        std::vector<Node> emptyNodesList(emptyNodes.begin(), emptyNodes.end());
+        Node nn1 = mkConcat(STRING_CONCAT, emptyNodesList);
+        Node res = nm->mkNode(kind::STRING_STRREPL, node[0], nn1, nn2);
+        return returnRewrite(node, res, "rpl-emp-cnts-substs");
```

Comment:
Do we ensure that res != node here?  What if y is a CONCAT?

Reply:
Good point, done.